### DISCUSSION TITLE
Future of DnD Kit?

### DISCUSSION THREAD

**Discussion by jonahallibone** [clauderic/dnd-kit#1156] [General]
Hi!

I recently wrote an application a couple months back using DnD Kit and it was a great experience. However, as my app has grown, I've noticed some performance issues with DnD Kit, as have others in the issues. I've even seen some PRs opened to address them, without responses. I'm just wondering if DnD kit is still being actively maintained, or if I should move along to another library for future development. I understand OSS can we incredibly thankless and I want to say i appeciate this project a lot. Any clarity on the future of DnD kit would be great!

Thanks

**Comment by pstrassmann** [comment]
Out of curiosity, what are the performance issues you're seeing?

**Reply by jonahallibone** [reply]
https://github.com/clauderic/dnd-kit/issues/1071

**Reply by pstrassmann** [reply]
Thanks! Yes, I have also noticed a limited ability to control the rerendering of components that use the `useDraggable` or `useSortable` hooks. Memoizing the "[presentational component](https://docs.dndkit.com/api-documentation/draggable/drag-overlay#presentational-components)" has helped us from incurring a noticeable performance hit, but, yes, I totally agree that it would be more ideal to have each component using those hooks be able to choose which of the hooks' return values to subscribe / be reactive to. Some of the patterns implemented by [zustand](https://github.com/pmndrs/zustand) come to mind.

But yes, overall I very much agree that it would be helpful to get a sense of what's on Clauderic's mind re his current plan for the future of the repo and input from other contributors. It is an incredible piece of work and, to me, very obviously the best we have in the React drag and drop world.

**Reply by jonahallibone** [reply]
The only viable alternative to dnd kit is react-dnd. I prefer the API design of Dnd kit much more, but in the event that this lib is unmaintained I fear I will have to switch.

@clauderic any possible updates on `dnd-kit` status?

**Reply by jonahallibone** [reply]
@pstrassmann btw: https://github.com/clauderic/dnd-kit/issues/1194
